### PR TITLE
Add setInputFiles docs for browser module

### PR DESCRIPTION
**Pull request by ankur22**
## What?

Add `setInputFiles` docs for browser module. The API is available in `frame`, `page` and `elementHandle`.

## Checklist

- [x] I have used a meaningful title for the PR.
- [x] I have described the changes I've made in the "What?" section above.
- [x] I have performed a self-review of my changes.
- [x] I have run the `npm start` command locally and verified that the changes look good.

- [x] I have made my changes in the `docs/sources/next` folder of the documentation.

## Related PR(s)/Issue(s)

https://github.com/grafana/xk6-browser/pull/1097 & https://github.com/grafana/xk6-browser/pull/1244

### CODE REVIEW

File: docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+| file.name           | string      | `''`    | The name of the file e.g. `file.txt`.                                                                                                                                                                                                                                                                                                         |
+| file.mimeType       | string      | `''`    | The type of the file content e.g. `text/plain`.                                                                                                                                                                                                                                                                                               |
+| file.buffer         | ArrayBuffer | `[]`    | base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md:17]:
```suggestion
| file.buffer         | ArrayBuffer | `[]`    | Base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+| Parameter           | Type        | Default | Description                                                                                                                                                                                                                                                                                                                                   |
+| ------------------- | ----------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| file                | object      | `null`  | This is a mandatory parameter.                                                                                                                                                                                                                                                                                                                |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md:14]:
```suggestion
| file                | object      | `null`  | This is a required parameter.                                                                                                                                                                                                                                                                                                                |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+              <input type="file" id="upload" multiple />
+              <input type="submit" value="Send" />
+          </form>
+      </body>
+    </html>`);
+
+  const eh = page.$('input[id="upload"]');
+
+  // The file is set to the input element with the id "upload".
+  eh.setInputFiles({
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md:62]: Where would this file have to be for this example to work?

File: docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+| ------------------- | ----------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| selector            | string      | `''`    | A selector to search for an element. If there are multiple elements satisfying the selector, the first will be used.                                                                                                                                                                                                                          |
+| file                | object      | `null`  | This is a mandatory parameter.                                                                                                                                                                                                                                                                                                                |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md:15]:
```suggestion
| file                | object      | `null`  | This is a required parameter.                                                                                                                                                                                                                                                                                                                |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+| file.name           | string      | `''`    | The name of the file e.g. `file.txt`.                                                                                                                                                                                                                                                                                                         |
+| file.mimeType       | string      | `''`    | The type of the file content e.g. `text/plain`.                                                                                                                                                                                                                                                                                               |
+| file.buffer         | ArrayBuffer | `[]`    | base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md:18]:
```suggestion
| file.buffer         | ArrayBuffer | `[]`    | Base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+| ------------------- | ----------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| file                | object      | `null`  | This is a mandatory parameter.                                                                                                                                                                                                                                                                                                                |
+| file.name           | string      | `''`    | The name of the file e.g. `file.txt`.                                                                                                                                                                                                                                                                                                         |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md:15]:
```suggestion
| file.name           | string      | `''`    | The name of the file. For example, `file.txt`.                                                                                                                                                                                                                                                                                                         |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+| file.mimeType       | string      | `''`    | The type of the file content e.g. `text/plain`.                                                                                                                                                                                                                                                                                               |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md:16]:
```suggestion
| file.mimeType       | string      | `''`    | The type of the file content. For example, `text/plain`.                                                                                                                                                                                                                                                                                               |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+| selector            | string      | `''`    | A selector to search for an element. If there are multiple elements satisfying the selector, the first will be used.                                                                                                                                                                                                                          |
+| file                | object      | `null`  | This is a mandatory parameter.                                                                                                                                                                                                                                                                                                                |
+| file.name           | string      | `''`    | The name of the file e.g. `file.txt`.                                                                                                                                                                                                                                                                                                         |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md:16]:
```suggestion
| file.name           | string      | `''`    | The name of the file. For example, `file.txt`.                                                                                                                                                                                                                                                                                                         |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+| file.mimeType       | string      | `''`    | The type of the file content e.g. `text/plain`.                                                                                                                                                                                                                                                                                               |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/frame/setinputfiles.md:17]:
```suggestion
| file.mimeType       | string      | `''`    | The type of the file content. For example, `text/plain`.                                                                                                                                                                                                                                                                                               |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+| ------------------- | ----------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| selector            | string      | `''`    | A selector to search for an element. If there are multiple elements satisfying the selector, the first will be used.                                                                                                                                                                                                                          |
+| file                | object      | `null`  | This is a mandatory parameter.                                                                                                                                                                                                                                                                                                                |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md:15]:
```suggestion
| file                | object      | `null`  | This is a required parameter.                                                                                                                                                                                                                                                                                                                |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+| file.name           | string      | `''`    | The name of the file e.g. `file.txt`.                                                                                                                                                                                                                                                                                                         |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md:16]:
```suggestion
| file.name           | string      | `''`    | The name of the file. For example, `file.txt`.                                                                                                                                                                                                                                                                                                         |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+| file.mimeType       | string      | `''`    | The type of the file content e.g. `text/plain`.                                                                                                                                                                                                                                                                                               |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md:17]:
```suggestion
| file.mimeType       | string      | `''`    | The type of the file content. For example, `text/plain`.                                                                                                                                                                                                                                                                                               |
```

File: docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+| file.buffer         | ArrayBuffer | `[]`    | base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/k6-experimental/browser/page/setinputfiles.md:18]:
```suggestion
| file.buffer         | ArrayBuffer | `[]`    | Base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
```

**Review by heitortsergent — COMMENTED** [review]
Just some minor style nits, and a question about the example. 🙏

**Review by inancgumus — APPROVED** [review]

File: docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md (hunk 3 above)
Comment (ankur22) [reply docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md:62]: The file is inline in the code i.e. it's all there in the object (`name`, `mimetype` and `buffer`). This new API doesn't work with the local filesystem at all. A user will need to work with the experimental FS module to do that.

**Comment by ankur22** [comment]
@heitortsergent i'm merging this into make the next release, but please add any more suggestions and I will add them to a new PR.

**Comment by heitortsergent** [comment]
@ankur22 I think it would be great if we had an example with a real use case, whether that's using the experimental `fs` module, or any other use case we think might be common.

What do you think? I could open a separate issue for it.

**Comment by ankur22** [comment]
> @ankur22 I think it would be great if we had an example with a real use case, whether that's using the experimental fs module, or any other use case we think might be common.

That's a good idea, and I already have one ready to be added into the docs. I'll create a PR soon.

**Comment by heitortsergent** [comment]
@ankur22 thank you!! 🙇

**Pull request merged**